openhtf/core/test_state.py

```diff
@@ -644,6 +644,17 @@ def _set_phase_outcome(self):
                  else test_record.PhaseOutcome.FAIL)
     self.phase_record.outcome = outcome
 
+  def _update_phase_result_from_phase_outcome(self):
```

Comment:
How about calling this _update_phase_result_from_failed_phase_outcome()  since it only updates the result for failed phases?